src/vaadin-combo-box-mixin.html

```diff
@@ -872,6 +872,7 @@
 
       if (this.selectedItem !== e.detail.item) {
         this.selectedItem = e.detail.item;
+        this._detectAndDispatchChange();
```

Comment:
adding this line here means that the following one has stopped working:
https://github.com/vaadin/vaadin-combo-box/blob/f3d2c44f65580e34a2b4c4087c95eb555f47484f/src/vaadin-combo-box-mixin.html#L599

Can we now remove that one and see if all the tests still pass?

Reply:
No, that one should still work for other cases, but here the problem (regression cause) was that now `close()` is called before the `selectedItem` is set so there are no changes at that point. See:
```js
      if (this.opened) {
        this.close();
      }
      if (this.selectedItem !== e.detail.item) {
        this.selectedItem = e.detail.item;
        this._detectAndDispatchChange();
      }
```

Reply:
But I will check if that other one could maybe be not needed anymore. I'll try running the tests without it, but I think it's still needed for when the overlay is closing is triggered in other ways.

Reply:
If that other `this._detectAndDispatchChange();` is removed then 3 tests fail.
![image](https://user-images.githubusercontent.com/75155/54811420-4f8b2a00-4c91-11e9-85e9-893245330b30.png)

Reply:
Ok, that's what I suspected but needed to clarify that.